### PR TITLE
crosscluster/logical: drop leases at each checkpoint

### PR DESCRIPTION
**Pull request by dt**
Release note: none.
Epic: none.

### CODE REVIEW

**Review by fqazi — APPROVED** [review]
:lgtm:

> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/134888#-:-OBSKChC5gIvLMt4ptQ3:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @msbutler)

File: pkg/ccl/crosscluster/logical/lww_row_processor.go [hunk 1]
```diff
@@ -68,7 +68,6 @@ type RowProcessor interface {
 	ProcessRow(context.Context, isql.Txn, roachpb.KeyValue, roachpb.Value) (batchStats, error)
 	GetLastRow() cdcevent.Row
 	SetSyntheticFailurePercent(uint32)
-	ReportMutations(*stats.Refresher)
```
Comment (msbutler) [inline pkg/ccl/crosscluster/logical/lww_row_processor.go:71]: was this unintentional?

Comment (dt) [reply pkg/ccl/crosscluster/logical/lww_row_processor.go:71]: Intentional -- it should have been in https://github.com/cockroachdb/cockroach/pull/134272 really but I was getting it while I was here.

**Review by msbutler — APPROVED** [review]

**Comment by dt** [comment]
TFTR!

bors r+

**Comment by msbutler** [comment]
blathers backport release-24.3.0-rc

**Pull request merged**